archive_verify/pdc_client.py

```diff
@@ -0,0 +1,141 @@
+import re
+import logging
+import os
+import subprocess
+import fnmatch
+
+# Share pre-configured workers log
+log = logging.getLogger('archive_verify.workers')
+
+
+class PdcClient():
+    """
+    Base class representing a PDC client.
+    Staging and production environments should instantiate PdcClient (default).
+    Local and testing environments should instantiate MockPdcClient.
+    """
+    def __init__(self, archive_name, archive_pdc_path, archive_pdc_description, job_id, config):
+        """
+        :param archive_name: The name of the archive we shall download
+        :param archive_pdc_path: The path in PDC TSM to the archive that we want to download
+        :param archive_pdc_description: The unique description that was used when uploading the archive to PDC
+        :param job_id: The current rq worker job id
+        :param config: A dict containing the apps configuration
+        """
+        self.dest_root = config["verify_root_dir"]
+        self.dsmc_log_dir = config["dsmc_log_dir"]
+        self.whitelisted_warnings = config["whitelisted_warnings"]
+        self.archive_name = archive_name
+        self.archive_pdc_path = archive_pdc_path
+        self.archive_pdc_description = archive_pdc_description
+        self.job_id = job_id
+
+    def dest(self):
+        """
+        :returns The unique path where the archive will be downloaded.
+        """
+        return "{}_{}".format(os.path.join(self.dest_root, self.archive_name), self.job_id)
+
+    def download(self):
```

Comment:
Maybe we should raise the log levels here to info? Feels like it could be interesting to see even when not debugging?

Reply:
Agree, fixing.